src/service/viewport-impl.js

```diff
@@ -784,6 +793,144 @@ export class ViewportBindingNaturalIosEmbed_ {
     documentBody.appendChild(this.endPosEl_);
 
     documentBody.addEventListener('scroll', this.onScrolled_.bind(this));
+
+    // DO NOT SUBMIT: only do this in canary at first.
+    // The fixed layer is a *sibling* of the body element. I.e. it's a direct
+    // child of documentElement. It's used to manage the `postition:fixed`
+    // elements in iOS-iframe case due to the
+    // https://bugs.webkit.org/show_bug.cgi?id=154399 bug, which is itself
+    // a result of workaround for the issue where scrolling is not supported
+    // in iframes (https://bugs.webkit.org/show_bug.cgi?id=149264).
+    // This implementation finds all elements that could be `fixed` and checks
+    // on major relayouts if they are indeed `fixed`. All `fixed` elements are
+    // moved into the fixed layer.
+    this.fixedLayer_ = this.win.document.createElement('div');
+    this.fixedLayer_.id = '-amp-fixedlayer';
+    setStyles(this.fixedLayer_, {
+      position: 'absolute',
+      top: 0,
+      left: 0,
+      right: 0,
+      bottom: 0,
+      pointerEvents: 'none'
+    });
+    this.win.document.documentElement.appendChild(this.fixedLayer_);
+
+    // Find all `position:fixed` elements.
+    const stylesheets = document.styleSheets;
+    const fixedSelectors = [];
+    for (let i = 0; i < stylesheets.length; i++) {
+      const stylesheet = stylesheets[i];
+      if (stylesheet.disabled ||
+              !stylesheet.ownerNode ||
+              stylesheet.ownerNode.tagName != 'STYLE' ||
+              stylesheet.ownerNode.hasAttribute('amp-boilerplate') ||
+              stylesheet.ownerNode.hasAttribute('amp-runtime') ||
+              stylesheet.ownerNode.hasAttribute('amp-extension')) {
+        continue;
+      }
+      this.discoverFixedSelectors_(stylesheet.cssRules, fixedSelectors);
+    }
+    fixedSelectors.forEach(selector => {
+      const elements = this.win.document.querySelectorAll(selector);
+      for (let i = 0; i < elements.length; i++) {
+        if (i > 10) {
```

Comment:
This is the proposal. This is a pretty heavy work involved, so there has to be a limit to how many elements are involved.